src/core/model/reducer.ts

```diff
@@ -72,10 +72,21 @@ export const uiSchemaReducer = (
         newUiSchema.elements.splice(action.index, 0, newUIElement);
         return getRoot(newUiSchema as EditorUISchemaElement);
       });
-    case SET_UISCHEMA_OPTIONS:
-      return withCloneTree(action.uiSchema, uiSchema, (newUiSchema) => {
-        newUiSchema.options = action.options;
-        return getRoot(newUiSchema as EditorUISchemaElement);
+    case UPDATE_UISCHEMA_ELEMENT:
+      return withCloneTree(action.uiSchemaElement, uiSchema, (newUiSchema) => {
+        const updatedUISchemaElement = {
+          ...newUiSchema,
+          ...action.changedProperties,
+        };
+
+        if (newUiSchema.parent && isEditorLayout(newUiSchema.parent)) {
+          const parent = newUiSchema.parent as EditorLayout;
+          const index = parent.elements.indexOf(newUiSchema);
+          // update element in its parent element list
+          parent.elements[index] = updatedUISchemaElement;
+        }
+
+        return getRoot(updatedUISchemaElement as EditorUISchemaElement);
```

Comment:
This works by miracle :smile: 

I was wondering why this worked for a group in a control as the parent is not updated here like for layouts. It turns out line 82-87 are without any effect.

The reason that this updates correctly is that JSON Forms in the properties view already creates a clone of the whole ui schema itself. This whole clone (with the correct parent and children elements set) is then given here as the `action.changedProperties`. With line `...action.changedProperties` the `updatedUISchemaElement` is basically again a clone as all of its properties will be replaced. We then return the element as the new ui schema.

This seems somewhat frickle ;) So I would like to change this to: (using the `merge` from lodash).
```suggestion
    case UPDATE_UISCHEMA_ELEMENT:
      return withCloneTree(action.uiSchemaElement, uiSchema, (newUiSchema) => {
        merge(newUiSchema, action.changedProperties);
        return getRoot(newUiSchema);
```

This has the same effect as the previous code.

Additionally I would like go back to the `Properties` and strip the data object given to JSON Forms from all our special properties and their parent and children, i.e. `uuid`, `parent`, `elements` and `options.detail`.